[TASK] #105297 - Deprecate tableoptions and collate connection config (#4875)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/Configuration/Typo3ConfVars/DB.rst

        Name of the database/schema to connect to. Can be used with
        MySQL/MariaDB and PostgreSQL.

    ..  confval:: defaultTableOptions
        :Path: $GLOBALS['TYPO3_CONF_VARS']['DB']['Connections'][<connection_name>]['defaultTableOptions']
        :name: typo3-conf-vars-db-connection-name-defaultTableOptions
        :type: array

>         Defines the charset and collation options for tables for MySQL/MariaDB:

Reviewer: ```suggestion
        Defines the charset and collation options when new tables are created (MySQL/MariaDB only):
```

Reviewer: Not sure about this exact wording, but we should IMO specify that 'defaultTableOptions' applies to CREATE TABLE statements, the charset here is not used for existing tables or the connection itself. What do you think?

Reviewer 2: I think your addition makes it more clear

        Name of the database/schema to connect to. Can be used with
        MySQL/MariaDB and PostgreSQL.

>     ..  _typo3ConfVars_db_connections_defaulttableoptions:

Reviewer 2: ```suggestion
```
It is not needed to introduce new anchors for confvals as the confval can already be referenced directly and the anchor would have no title. We only kept the old anchors for backward compability in case they have already been referenced

        :type: array
        :Default: []

        ..  deprecated:: 13.4
            Since TYPO3 v11 the :php:`tableoptions` keys were silently migrated
            to :confval:`typo3-conf-vars-db-connection-name-defaultTableOptions`,
            which is the proper Doctrine DBAL connection option for MariaDB and
            MySQL.

>             Furthermore, Doctrine DBAL 3.x switched from using they array key

Reviewer: ```suggestion
            Furthermore, Doctrine DBAL 3.x switched from using the array key
```
